src/Attestation/Attestation.Test/ScenarioTests/AttestationTestRunner.cs

```diff
@@ -0,0 +1,100 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Attestation;
+using Microsoft.Azure.Management.Attestation;
+using Microsoft.Azure.Test.HttpRecorder;
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Microsoft.Azure.Commands.TestFx;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Commands.Attestation.Test.ScenarioTests
+{
+    public class AttestationTestRunner
+    {
+        protected readonly ITestRunner TestRunner;
+
+        protected AttestationTestRunner(ITestOutputHelper output)
+        {
+            TestRunner = TestManager.CreateInstance(output)
+                .WithNewPsScriptFilename($"{GetType().Name}.ps1")
+                .WithProjectSubfolderForTests("ScenarioTests")
+                .WithCommonPsScripts(new[]
+                {
+                    @"Common.ps1",
+                    @"../AzureRM.Resources.ps1"
+                })
+                .WithNewRmModules(helper => new[]
+                {
+                    helper.RMProfileModule,
+                    helper.GetRMModulePath("Az.Attestation.psd1")
+                })
+                .WithNewRecordMatcherArguments(
+                    userAgentsToIgnore: new Dictionary<string, string>
+                    {
+                        {"Microsoft.Azure.Management.Resources.ResourceManagementClient", "2016-02-01"},
+                        {"Microsoft.Azure.Management.ResourceManager.ResourceManagementClient", "2017-05-10"}
+                    },
+                    resourceProviders: new Dictionary<string, string>
+                    {
+                        {"Microsoft.Resources", null},
+                        {"Microsoft.Features", null},
+                        {"Microsoft.Authorization", null}
+                    }
+                ).WithManagementClients(
+                    GetResourceManagementClient,
```

Comment:
Without WithManagementClients will set TestClientFactory instead of the MockClientFactory which resulted the issue. May need to remove this line of code after rolling out the new testfx code.